src/components/ClassroomComponent/PublicityPermComponent/Publicity.js

```diff
@@ -0,0 +1,266 @@
+        booker_contact: '9898989898',
+        booker_reg_no: '150911111'
+      },
+      event_fields: {
+        title: '',
+        start_date: null,
+        end_date: null
+      },
+      SnackBarmessage: '',
+      openSnackBar: false,
+      autoHideDuration: 3000,
+      indexes: [{ '0': false, '1': false, '2': false, '3': false },
+        { '0': false, '1': false, '2': false, '3': false },
+        { '0': false, '1': false, '2': false, '3': false },
+        { '0': false, '1': false, '2': false, '3': false }],
+      files: [],
+      bookedEvent: null,
+      disableSubmit: false
+
+    }
+  };
+
+  componentWillMount () {
+    this.setState({
+      checked: [false, false, false, false]
+    })
+  }
+
+  handleNext () {
+    if (this.state.stepIndex < 2) {
+      const { stepIndex } = this.state
+      this.setState({ stepIndex: stepIndex + 1 })
+    }
+    if (this.state.stepIndex === 2) { this.handleSubmit() }
+  };
+
+  handlePrev () {
+    const { stepIndex } = this.state
+    if (stepIndex > 0) {
+      this.setState({ stepIndex: stepIndex - 1, isFormValid: true })
+    }
+  };
+
+  handleSnackBarClose () {
+    this.setState({ openSnackBar: false })
+  }
+
+  handleSubmit () {
+    this.setState({ disableSubmit: true })
+    var result = this.parseMediums()
+    var files = this.state.files
+    var newData = {
+      'AD_appr': this.props.user.isSC ? 'pending' : 'NA',
+      'FA_appr': this.props.user.isSC ? 'approved' : 'pending',
+      'SO_appr': 'NA',
+      'clubName': this.props.user.name,
+      'clubID': this.props.user.uid,
+      'FA_name': this.props.user.fa.name,
+      'FA_date': this.props.user.isSC ? moment(this.state.today, 'DD-MM-YYYY').format('DD-MM-YYYY') : null
+    }
+    var bookerFields = {
+      'bookerFields': this.state.bookerFields
+    }
+    var fields = this.state.event_fields
+    fields['start_date'] = moment(this.state.event_fields['start_date'], 'DD-MM-YYYY').format('DD-MM-YYYY')
+    fields['end_date'] = moment(this.state.event_fields['end_date'], 'DD-MM-YYYY').format('DD-MM-YYYY')
+
+    newData = Object.assign({}, newData, bookerFields, fields)
+    var publicityID = newData.clubID.slice(0, 4)
+    publicityID = publicityID.concat(this.state.event_fields['title'].toLowerCase().slice(0, 4))
+    publicityID = publicityID.concat(new Date().getTime() % 1000000)
+    var obj = Object.assign({}, result, newData)
+
+    Promise.all(files.map(file => storage.ref().child(`uid/${publicityID}/${file.name}`).put(file).then(response => response.downloadURL))).then(
+      urls => {
+        obj['files'] = urls
+        this.handleDBSubmit(obj, publicityID)
+      })
+  }
+
+  handleDBSubmit (obj, publicityID) {
+    firebaseDB.ref('/publicity/').child(publicityID).set(obj)
+    var scope = this
+    firebaseDB.ref('/users/' + scope.props.user.uid + '/my_publicity/').push(publicityID,
+      function (res, err) {
+        if (err) { console.log("couldn't be booked ", err) } else {
```

Comment:
Handle the error in some way. Just refresh the page maybe or display proper toast message. User cannot see the console.logs